src/realm/object-store/shared_realm.hpp

```diff
@@ -674,6 +674,14 @@ class InvalidEncryptionKeyException : public std::logic_error {
     {
     }
 };
+
+class InvalidConversationForFlxSyncRealm : public std::logic_error {
+public:
+    InvalidConversationForFlxSyncRealm()
+        : std::logic_error("Realm cannot be converted if flx sync is enabled")
```

Comment:
"cannot be converted if flexible sync is enabled" - "flx" is an acronym we expose in some places, but we should probably try to use the full name in error messages, just to make things clear for users.